### ISSUE TITLE
Is it possible for D-RISE to generate heatmaps for mask R-CNN?

### ISSUE THREAD

**Issue by mingqiJ** [RuoyuChen10/objectdetection-saliency-maps#4]
Hi, I see you use **Yolo v3** to do many tests. Is it possible to apply your D-RISE to the **mask R-CNN**?

**Comment by RuoyuChen10** [comment]
> Hi, I see you use **Yolo v3** to do many tests. Is it possible to apply your D-RISE to the **mask R-CNN**?

Yes, you can have a try, load mask-cnn config, it should does work.

**Comment by mingqiJ** [comment]
Okay. Thanks for your reply. 
You mean maybe directly use mask-cnn it can work? I will try it.

**Comment by RuoyuChen10** [comment]
> Okay. Thanks for your reply. You mean maybe directly use mask-cnn it can work? I will try it.

Yes, the only possible note is that Mask R-CNN will return masks, maybe a little note because methods like yolo v3 will only return boxes.

**Comment by mingqiJ** [comment]
Hi, Is the D-RISE very slow? When I use 5000 nmask, use Mask R-CNN. It takes about 220s each object. Very slow

**Comment by RuoyuChen10** [comment]
> Hi, Is the D-RISE very slow? When I use 5000 nmask, use Mask R-CNN. It takes about 220s each object. Very slow

Yes, in the paper, D-RISE runs in approximately 70s per image (for all detections) for YOLOv3 and 170s for Faster R-CNN on NVidia Tesla V100.

**Comment by mingqiJ** [comment]
Yes, I also see it. So slow. Very confused about how did they finish all MS COCO valid images if they use Faster R-CNN.....

**Comment by RuoyuChen10** [comment]
> Yes, I also see it. So slow. Very confused about how did they finish all MS COCO valid images if they use Faster R-CNN.....

When inference, each image can only choose one proposal for evaluation (Point Game).

**Comment by mingqiJ** [comment]
You mean they only use 1 object for each image to evaluate (Point Game)?

**Comment by RuoyuChen10** [comment]
> You mean they only use 1 object for each image to evaluate (Point Game)?

One proposal has the highest prediction score.
